chartfx-chart/src/main/java/de/gsi/chart/renderer/Renderer.java

```diff
@@ -67,4 +67,14 @@
      * @return true (default) if data sets are supposed to be drawn
      */
     BooleanProperty showInLegendProperty();
+
+    /**
+     * Indicates whether the renderer supports dataset visibility.
+     *
+     * @return false (default) if the renderer does not support visibility.
+     */
+    default boolean supportsVisibility() {
```

Comment:
as commented above... possibly this could be removed and made a default requirements for all renderer